topics/imaging/tutorials/tutorial-CP/tutorial.md

```diff
@@ -0,0 +1,633 @@
+
+> ### {% icon hands_on %} Hands-on: Segment nuclei that are complete within the boundaries of the image
+>
+> 1. **IdentifyPrimaryObjects** {% icon tool %} with the following parameters:
+>    - {% icon param-file %} *"Select the input CellProfiler pipeline"*: `output_pipeline` (output of **Starting Modules** {% icon tool %})
+>    - *"Use advanced settings?"*: `Yes, use advanced settings`
+>        - *"Enter the name of the input image (from NamesAndTypes)"*: `DNA`
+>        - *"Enter the name of the primary objects to be identified"*: `Nuclei`
+>        - *"Typical minimum diameter of objects, in pixel units (Min)"*: `15`
+>        - *"Typical maximum diameter of objects, in pixel units (Max)"*: `200`
+>        - *"Discard objects outside the diameter range?"*: `Yes`
+>        - *"Discard objects touching the border of the image?"*: `Yes`
+>        - *"Threshold strategy"*: `Global`
+>            - *"Thresholding method"*: `Otsu`
+>                - *"Two-class or three-class thresholding?"*: `Two classes`
+>                - *"Threshold correction factor"*: `0.9`
+>        - *"Method to distinguish clumped objects"*: `Shape`
+>            - *"Method to draw dividing lines between clumped objects"*: `Shape`
+>                - *"Automatically calculate size of smoothing filter for declumping?"*: `Yes`
+>                - *"Automatically calculate minimum allowed distance between local maxima?"*: `Yes`
+>        - *"Handling of objects if excessive number of objects identified"*: `Continue`
+>
+>
+>
+>    > ### {% icon comment %} Comment
+>    >
+>    > - The name entered to the input image and objects has to be consistent (case sensitive) with the names in *NamesAndTypes* and the tools to follow.
+>    > - The min and max diameter of the objects (`Typical minimum diameter of objects, in pixel units (Min)` and `Typical minimum diameter of objects, in pixel units (Max)`) will have to be adjusted to the resolution of the images.
+>    {: .comment}
+>
+{: .hands_on}
+
+> ### {% icon question %} Questions
+>
+> We are using here Otsu's method for segmentation. What other segmentation options are available? What is the difference between them?
+>
+> > ### {% icon solution %} Solution
+> >
+> > In the *global* methods we have `Manual`, `Measurement`, `Minimum cross entropy`, `Otsu` and `Robust background`. For the *adaptive* ones we only have `Otsu`. Check the parameters' help to get more information on each one. 
+> >
+> {: .solution}
+>
+{: .question}
+
+From the previous tool, we got a group of objects (nuclei). Now, we want to export the segmentation masks as a single image to check how well the segmentation algorithm is performing. We also want to label the nuclei with their identifiers for future visual inspection of the results. The output of this step will look like:
+
+![Identified nuclei with labels](../../images/tutorial-CP/img_nuclei_labels.png "Identified nuclei with labels.")
```

Comment:
I don't know if it would be possible but it would be nice to visualize what we do (and not wait for the very end).

Reply:
No, it's not possible :( The tools are creating a pipeline file but not running any analysis until the end when the file is completely assembled. I'm mentioning it explicitly in the text now so that people don't expect to see the image at that point.